common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -380,16 +380,19 @@ public static boolean deviceModeFromBundle(@NonNull final Bundle bundle) {
     }
 
     /**
-     * Get a bundle from current account's AccountRecord.
-     *
-     * @param record current account's AccountRecord.
+     * Get a bundle from current account's List<ICacheRecord>.
+     * 
+     * @param cacheRecords current account's List<ICacheRecord>.
      * @return Bundle
      */
-    public static Bundle bundleFromCurrentAccount(@NonNull final AccountRecord record) {
+    public static Bundle bundleFromCurrentAccount(@NonNull final List<ICacheRecord> cacheRecords) {
         final Bundle resultBundle = new Bundle();
 
-        final String recordInGson = new Gson().toJson(record, AccountRecord.class);
-        resultBundle.putString(BROKER_CURRENT_ACCOUNT, recordInGson);
+        if (cacheRecords != null) {
+            final Type listOfCacheRecords = new TypeToken<List<ICacheRecord>>(){}.getType();
```

Comment:
For @heidijinxujia: Your PR does something very similar [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/515/files#diff-44f2260eafcc843207898d147d7285beR326) -- rather than individually serialize _each_ `ICacheRecord` can you instead use Dome's trick of serializing the whole `List`?